src/app/vault/add-edit.component.ts

```diff
@@ -212,6 +214,20 @@ export class AddEditComponent implements OnChanges {
         document.getElementById('loginPassword').focus();
     }
 
+    async checkPassword() {
+        this.analytics.eventTrack.next({ action: 'Check Password' });
```

Comment:
should you check and return early if `this.cipher.login == null || this.cipher.login.password == null || this.cipher.login.password === ''`?